Source/Server/Packets/ChangeTeam.c

```diff
@@ -4,24 +4,34 @@
 
 void receive_change_team(server_t* server, player_t* player, stream_t* data)
 {
+    // @Todo: if player is not initialised yet (pyspades uses if not self.name) ignore this
+
     uint8_t received_id = stream_read_u8(data);
-    server->protocol.num_team_users[player->team]--;
-    uint8_t team = stream_read_u8(data);
 
-    if (team == 255) {
-        team = 2;
-    }
+    uint8_t team     = stream_read_u8(data);
+    uint8_t old_team = player->team;
+
+    if (old_team == team)
+        return;
 
-    if (team != 0 && team != 1 && team != 2) {
+    if (team != TEAM_A && team != TEAM_B && team != TEAM_SPECTATOR) {
         LOG_WARNING("Player %s (#%hhu) sent invalid team. Switching them to Spectator", player->name, player->id);
-        team = 2;
+        team = TEAM_SPECTATOR;
     }
 
-    player->team = team;
     if (player->id != received_id) {
         LOG_WARNING("Assigned ID: %d doesnt match sent ID: %d in change team packet", player->id, received_id);
     }
-    server->protocol.num_team_users[player->team]++;
-    send_kill_action_packet(server, player, player, 5, 5, 0);
-    player->state = STATE_WAITING_FOR_RESPAWN;
+
+    if (server->protocol.num_team_users[old_team] > 0)
+        server->protocol.num_team_users[old_team]--;
```

Comment:
Previous behaviour resulted in
num_team_users {
TEAM_A: 1
TEAM_B: 2
TEAM_SPECTATOR: 255, 254, 253....
}